p2p/log.go

```diff
@@ -9,100 +14,131 @@ const (
 	MtDatasetLog = MsgType("dataset_log")
 )
 
+// DatasetLogRequestBody encapsulates options for requesting dataset history
```

Comment:
Agree we need this struct, name feels a little verbose. Maybe just `DatasetLogRequest`?